src/main/java/uk/gov/register/core/RegisterContextFactory.java

```diff
@@ -13,19 +13,25 @@
 import javax.validation.Valid;
 import javax.validation.constraints.NotNull;
 import java.util.Collections;
+import java.util.Optional;
 
 public class RegisterContextFactory {
     @Valid
     @NotNull
     @JsonProperty
     private DataSourceFactory database;
 
+    @Valid
+    @JsonProperty
+    private String trackingId;
+
     @SuppressWarnings("unused, used by Jackson")
     public RegisterContextFactory() {
     }
 
-    public RegisterContextFactory(DataSourceFactory database) {
+    public RegisterContextFactory(DataSourceFactory database, Optional<String> trackingId) {
         this.database = database;
+        this.trackingId = trackingId.orElse(null);
```

Comment:
I couldn't get this to work with Jackson, sorry. If you know how, happy to make the change